peerprovider/interface.go

```diff
@@ -0,0 +1,46 @@
+package peerprovider
+
+import (
+	"fmt"
+	"net/url"
+
+	"golang.org/x/net/context"
+)
+
+var registry = make(map[string]PeerProvider)
+
+func init() {
+	RegisterPeerProvider("file", filePeerProvider{})
+}
+
+// Schemes returns supported peer provider protocol schemes.
```

Comment:
Can we add unit tests in this package for:
 - registering a new scheme
 - making sure it's returned in `Schemes`
 - ensuring it's used when `Resolve` is called with that scheme